InstructorSubmissionAdjustmentUiTest: improve success rate against live server

Usually this passes after several tries against list server. At v5.104, it failed even after 5 tries.
```
java.lang.AssertionError: 
Expected :0
Actual   :1
	at org.testng.AssertJUnit.fail(AssertJUnit.java:59)
	at org.testng.AssertJUnit.failNotEquals(AssertJUnit.java:364)
	at org.testng.AssertJUnit.assertEquals(AssertJUnit.java:80)
	at org.testng.AssertJUnit.assertEquals(AssertJUnit.java:245)
	at org.testng.AssertJUnit.assertEquals(AssertJUnit.java:252)
	at teammates.test.cases.BaseTestCase.assertEquals(BaseTestCase.java:141)
	at teammates.test.cases.browsertests.InstructorSubmissionAdjustmentUiTest.testAdjustmentOfSubsmission(InstructorSubmissionAdjustmentUiTest.java:83)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:659)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:72)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:127)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
```

--- a/src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java
+++ b/src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java
@@ -80,6 +80,9 @@ public class InstructorSubmissionAdjustmentUiTest extends BaseUiTestCase {
                      + student.toEnrollmentString();
         enrollPage.enroll(enrollString);
 
+        // It might take a while for the submission adjustment to persist (especially on the live server),
+        // during which the pre-existing submissions and responses would be counted.
+        // Hence, this needs to be retried several times until the count becomes zero.
         RetryManager.runUntilSuccessful(new RetryableTaskReturns<Integer>("Assert zero new responses") {
             @Override
             public Integer run() {

--- a/src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java
+++ b/src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java
@@ -16,6 +16,8 @@ import teammates.common.util.ThreadHelper;
 import teammates.test.driver.BackDoor;
 import teammates.test.driver.Priority;
 import teammates.test.driver.TestProperties;
+import teammates.test.driver.retry.RetryManager;
+import teammates.test.driver.retry.RetryableTaskReturns;
 import teammates.test.pageobjects.InstructorCourseEnrollPage;
 
 /**
@@ -63,8 +65,8 @@ public class InstructorSubmissionAdjustmentUiTest extends BaseUiTestCase {
         ______TS("typical case : existing student changes team");
         loadEnrollmentPage();
 
-        FeedbackSessionAttributes session = testData.feedbackSessions.get("session2InCourse1");
-        StudentAttributes student = testData.students.get("student1InCourse1");
+        final FeedbackSessionAttributes session = testData.feedbackSessions.get("session2InCourse1");
+        final StudentAttributes student = testData.students.get("student1InCourse1");
 
         //Verify pre-existing submissions and responses
         List<FeedbackResponseAttributes> oldResponsesForSession =
@@ -78,9 +80,17 @@ public class InstructorSubmissionAdjustmentUiTest extends BaseUiTestCase {
                      + student.toEnrollmentString();
         enrollPage.enroll(enrollString);
 
-        int numberOfNewResponses =
-                getAllResponsesForStudentForSession(student, session.getFeedbackSessionName()).size();
-        assertEquals(0, numberOfNewResponses);
+        RetryManager.runUntilSuccessful(new RetryableTaskReturns<Integer>("Assert zero new responses") {
+            @Override
+            public Integer run() {
+                return getAllResponsesForStudentForSession(student, session.getFeedbackSessionName()).size();
+            }
+
+            @Override
+            public boolean isSuccessful(Integer result) {
+                return result == 0;
+            }
+        });
 
     }
 

--- a/src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java
+++ b/src/test/java/teammates/test/cases/browsertests/InstructorSubmissionAdjustmentUiTest.java
@@ -83,15 +83,15 @@ public class InstructorSubmissionAdjustmentUiTest extends BaseUiTestCase {
         // It might take a while for the submission adjustment to persist (especially on the live server),
         // during which the pre-existing submissions and responses would be counted.
         // Hence, this needs to be retried several times until the count becomes zero.
-        RetryManager.runUntilSuccessful(new RetryableTaskReturns<Integer>("Assert zero new responses") {
+        RetryManager.runUntilSuccessful(new RetryableTaskReturns<Integer>("Assert outdated responses removed") {
             @Override
             public Integer run() {
                 return getAllResponsesForStudentForSession(student, session.getFeedbackSessionName()).size();
             }
 
             @Override
-            public boolean isSuccessful(Integer result) {
-                return result == 0;
+            public boolean isSuccessful(Integer numberOfResponses) {
+                return numberOfResponses == 0;
             }
         });
 

